src/node_messaging.h

```diff
@@ -144,14 +198,9 @@ class MessagePortData : public TransferData {
   // This mutex protects all fields below it, with the exception of
   // sibling_.
   mutable Mutex mutex_;
-  std::list<Message> incoming_messages_;
+  std::list<std::shared_ptr<Message>> incoming_messages_;
```

Comment:
It might make sense to turn this into a `std::deque`/`std::vector`, `std::list` was nice because `Message` is a potentially large object and a linked list provided referential integrity, but with `std::shared_ptr`, that concern goes away